lib/xebow/keys.ex

```diff
@@ -102,20 +119,65 @@ defmodule Xebow.Keys do
     poll_timer_ms = 15
     :timer.send_interval(poll_timer_ms, self(), :update_pin_values)
 
+    animations =
+      Animation.types()
+      |> Enum.map(&Animation.new(type: &1))
+
     {:ok,
      %{
        pins: pins,
        keyboard_state: keyboard_state,
-       hid: hid
+       hid: hid,
+       animations: animations,
+       current_animation_index: 0
      }}
   end
 
+  @impl GenServer
+  def handle_cast(:next_animation, state) do
+    next_index = state.current_animation_index + 1
+
+    next_index =
+      case next_index < Enum.count(state.animations) do
+        true -> next_index
+        _ -> 0
+      end
+
+    animation = Enum.at(state.animations, next_index)
+
+    Xebow.Engine.play_animation(animation)
+
+    state = %{state | current_animation_index: next_index}
+
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_cast(:previous_animation, state) do
+    previous_index = state.current_animation_index - 1
+
+    previous_index =
+      case previous_index < 0 do
+        true -> Enum.count(state.animations) - 1
+        _ -> previous_index
+      end
+
+    animation = Enum.at(state.animations, previous_index)
+
+    Xebow.Engine.play_animation(animation)
+
+    state = %{state | current_animation_index: previous_index}
+
+    {:noreply, state}
+  end
+
   @impl GenServer
   def handle_info({:hid_report, hid_report}, state) do
     IO.binwrite(state.hid, hid_report)
     {:noreply, state}
   end
 
+  @impl GenServer
```

Comment:
I thought you only needed the `@impl` on the first function body.

Reply:
I'm fine either way. With @vanvoljg's approach it's harder to forget to add it back in if one is removed during refactoring.

Reply:
I thought it mostly made it clearer that it's a callback. I can omit in the future them if they make things too cluttered.